### PR TITLE
cmd/atlas/apply: without printing diff

### PR DESCRIPTION
**Pull request by yonidavidson**

### CODE REVIEW

File: cmd/atlas/mux.go [hunk 1]
```diff
@@ -23,7 +24,17 @@ type (
 		schema.Differ
 		schema.Execer
 		schema.Inspector
-		MarshalSpec func(v interface{}, marshaler schemaspec.Marshaler) ([]byte, error)
+		MarshalSpec   func(v interface{}, marshaler schemaspec.Marshaler) ([]byte, error)
+		UnMarshalSpec func(data []byte, unmarshaler schemaspec.Unmarshaler, v interface{}) error
```
Comment (rotemtam) [inline cmd/atlas/mux.go:28]:
```suggestion
		UnmarshalSpec func(data []byte, unmarshaler schemaspec.Unmarshaler, v interface{}) error
```

File: cmd/atlas/mux.go [hunk 2]
```diff
@@ -23,7 +24,17 @@ type (
...
+	}
+
+	schemaUnmarshal struct {
+		unmarshalSpec func(data []byte, unmarshaler schemaspec.Unmarshaler, v interface{}) error
+		unmarshaler   schemaspec.Unmarshaler
+	}
+
+	schemaUnmarshaler interface {
```
Comment (rotemtam) [inline cmd/atlas/mux.go:36]: why is this interface needed?

File: cmd/atlas/apply.go [hunk 3]
```diff
@@ -29,3 +39,27 @@ func init() {
...
+	err = u.unmarshal(f, &desired)
+	changes, err := d.Differ.SchemaDiff(s, &desired)
+	cobra.CheckErr(err)
+	prompt := promptui.Select{
+		Label: "Are you sure?",
+		Items: []string{"Continue", "Abort"},
+	}
+	_, result, err := prompt.Run()
+	cobra.CheckErr(err)
+	if result == "Continnue" {
```
Comment (rotemtam) [inline cmd/atlas/apply.go:61]: typo?

File: cmd/atlas/apply.go [hunk 4]
```diff
@@ -29,3 +39,27 @@ func init() {
...
+		err = d.Execer.Exec(ctx, changes)
```
Comment (rotemtam) [inline cmd/atlas/apply.go:62]: d.Exec

**Review by rotemtam — CHANGES_REQUESTED** [review]

File: cmd/atlas/apply.go (hunk 4 above)
Comment (yonidavidson) [reply cmd/atlas/apply.go:67]: OK, removed the DB the driver has since we don't need it at the moment.

File: cmd/atlas/mux.go [hunk 5]
```diff
@@ -93,10 +102,14 @@ func schemaNameFromDSN(url string) (string, error) {
 		}
 		return cfg.Database, err
 	default:
-		return "", fmt.Errorf("unknown database type: %q", key)
+		return "", errors.New("failed to get DB name from connection")
```
Comment (rotemtam) [inline cmd/atlas/mux.go:105]: ?

**Comment by yonidavidson** [comment]
good catch, missed on the merge.

**Pull request merged**